returnn/config.py

```diff
@@ -131,6 +131,40 @@ def get(self, key, default=None):
             return self.post_config[key]
         return self.config.get(key, default)
 
+    def update(self, other):
+        """
+        updates a ReturnnConfig with an other ReturnnConfig
```

Comment:
I think we should write here how each component is updated.

Reply:
```suggestion
        updates a ReturnnConfig with an other ReturnnConfig:
          * config, post_config, and pprint_kwargs use dict.update
          * prolog, epilog, and hashes are concatenated
          * staged_network_dict, sort_config, and black_formatting are overwritten
```
pending our discussion about `staged_network_dict`